contracts/TradeTypesLib2.sol

```diff
@@ -0,0 +1,134 @@
+/*
+ * Hubii Nahmii
+ *
+ * Compliant with the Hubii Nahmii specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity >=0.4.25 <0.6.0;
+
+import {MonetaryTypesLib} from "./MonetaryTypesLib.sol";
+import {NahmiiTypesLib} from "./NahmiiTypesLib.sol";
+
+/**
+ * @title     TradeTypesLib
+ * @dev       Data types centered around trade
+ */
+library TradeTypesLib2 {
+    //
+    // Enums
+    // -----------------------------------------------------------------------------------------------------------------
+    enum CurrencyRole {Intended, Conjugate}
+    enum LiquidityRole {Maker, Taker}
+    enum Intention {Buy, Sell}
+    enum TradePartyRole {Buyer, Seller}
+
+    //
+    // Structures
+    // -----------------------------------------------------------------------------------------------------------------
+    struct Operator {
+        uint256 id;
+        string data;
+    }
+
+    struct OrderPlacement {
+        Intention intention;
+
+        int256 amount;
```

Comment:
Signed amount?

Reply:
Yes. Monetary amounts are signed throughout. Certain monetary mounts like settled balances in _BalanceTracker_ may have negative values. And in Solidity `uint256` and `int256` are not implicitly convertible.